src/adapters/http/client.ts

```diff
@@ -0,0 +1,72 @@
+import Adapter from "../../lib/adapter.js"
+import GleeMessage from "../../lib/message.js"
+import got from "got"
+import { HttpAuthConfig, HttpAdapterConfig } from "../../lib/index.js"
+import http from "http"
+class HttpClientAdapter extends Adapter {
+
+  name(): string {
+    return "HTTP client"
+  }
+  async connect(): Promise<this> {
+    return this._connect()
+  }
+
+  async send(message: GleeMessage): Promise<void> {
+    return this._send(message)
+  }
+
+  async _connect(): Promise<this> {
+    this.emit("connect", {
+      name: this.name(),
+      adapter: this,
+      connection: http,
+      channel: this.channelNames,
+    })
+    return this
+  }
+  async _send(message: GleeMessage): Promise<void> {
+    const headers = {}
+    const config: HttpAdapterConfig = await this.resolveProtocolConfig("http")
+    const auth: HttpAuthConfig = await this.getAuthConfig(config.client.auth)
+    headers["Authentication"] = auth?.token
+    const serverUrl = this.serverUrlExpanded
+    this.channelNames.forEach(async (channelName) => {
+      const channelInfo = this.parsedAsyncAPI.channel(channelName)
+      const httpChannelBinding = channelInfo.binding("http")
+      const servers = channelInfo.servers()
+      if (
+        httpChannelBinding &&
+        (!servers.length || servers.includes(message.serverName))
+      ) {
+        const method = httpChannelBinding.method //get post put
```

Comment:
```suggestion
        const method = httpChannelBinding.method
```

There are many more methods that GET, POST, and PUT.